WORKSPACE

```
@@ -153,64 +153,47 @@ llvm_zlib_system(name = "zlib")
 # Flex/Bison rules
 ###############################################################################
 
-rules_m4_version = "0.2.1"
+rules_m4_version = "0.2.3"
 
 http_archive(
     name = "rules_m4",
-    patch_args = ["-p1"],
-    patches = [
-        # Trying to upstream: https://github.com/jmillikin/rules_m4/pull/15
-        "@carbon//bazel/patches/m4:0001_Support_M4_building_on_FreeBSD.patch",
-    ],
-    sha256 = "eaa674cd84546038ecbcc49cdd346134a20961a41fa1a541e80d8bf4b470c34d",
-    strip_prefix = "rules_m4-{0}".format(rules_m4_version),
-    urls = ["https://github.com/jmillikin/rules_m4/archive/v{0}.tar.gz".format(rules_m4_version)],
+    sha256 = "10ce41f150ccfbfddc9d2394ee680eb984dc8a3dfea613afd013cfb22ea7445c",
+    urls = ["https://github.com/jmillikin/rules_m4/releases/download/v{0}/rules_m4-v{0}.tar.xz".format(rules_m4_version)],
 )
 
 load("@rules_m4//m4:m4.bzl", "m4_register_toolchains")
 
 # When building M4, disable all compiler warnings as we can't realistically fix
 # them anyways.
-m4_register_toolchains(extra_copts = ["-w"])
+m4_register_toolchains(version = "1.4.18", extra_copts = ["-w"])
 
-# TODO: Can switch to a normal release version when it includes:
-# https://github.com/jmillikin/rules_flex/commit/1f1d9c306c2b4b8be2cb899a3364b84302124e77
-rules_flex_version = "1f1d9c306c2b4b8be2cb899a3364b84302124e77"
+rules_flex_version = "0.2.1"
 
 http_archive(
     name = "rules_flex",
-    sha256 = "a4e99a0a241c8a5aa238e81724ea3529722522c3702fd3aa674add5eb9807002",
-    strip_prefix = "rules_flex-{0}".format(rules_flex_version),
-    urls = ["https://github.com/jmillikin/rules_flex/archive/{0}.tar.gz".format(rules_flex_version)],
+    sha256 = "8929fedc40909d19a4b42548d0785f796c7677dcef8b5d1600b415e5a4a7749f",
+    urls = ["https://github.com/jmillikin/rules_flex/releases/download/v{0}/rules_flex-v{0}.tar.xz".format(rules_flex_version)],
 )
 
 load("@rules_flex//flex:flex.bzl", "flex_register_toolchains")
 
 # When building Flex, disable all compiler warnings as we can't realistically
 # fix them anyways.
-flex_register_toolchains(extra_copts = ["-w"])
+flex_register_toolchains(version = "2.6.4", extra_copts = ["-w"])
```

Same as above, let's continue to pick up the default:

```suggestion
flex_register_toolchains(extra_copts = ["-w"])
```